testing/src/scenario/state.py

```diff
@@ -110,6 +112,26 @@
 }
 
 
+def _detach_dicts_and_lists(cls):
```

Comment:
The name "detach" seems a bit odd to me -- why not `_copy_dicts_and_lists`? Or perhaps `_deepcopy_mutable_fields`?